app/api/views.py

```diff
@@ -150,14 +150,40 @@ def getspaces():
 @login_required
 def addproducts(current_user):
     data = request.get_json()
-    new_product = Product(name=data['productName'],description=data['productDesc'],sale_unit=data['sale_unit'],price=data['price'],images=data['images'],Instock=data['available_stock'],discount=data['discount'],date_created=d.datetime.utcnow())
+    new_product = Product(name=data['productName'],description=data['productDesc'],sale_unit=data['sale_unit'],price=data['price'],images=data['images'],Instock=data['available_stock'],discount=data['discount'],date_created=d.datetime.utcnow(), productID=str(uuid4()))
     db.session.add(new_product)
     # fetch store 
     current_space = current_user.space
     new_product.space = current_space
     db.session.commit()
     return jsonify({'msg':f'Added {new_product.name} successfully'}),200
 
+@api.route('/getallproducts', methods=['GET'])
+@login_required
+def getallproducts(current_user):
+    products = Product.query.all();
+    return jsonify(products_schema.dump(products)), 200
+
+@api.route('/updateproduct/<product_id>', methods=['POST'])
+@login_required
+def updateproduct(current_user, product_id):
+    data = request.get_json(force=True) # route accepts json
+    product = Product.query.filter_by(productID=product_id).first()
+
+    product.name = data['productName']
+    product.description = data['productDesc']
+    product.price = data['price']
+    product.images = data['images']
+    product.Instock = data['available_stock']
+    product.discount = data['discount']
+    product.date_created = d.datetime.utcnow()
+
+    current_space = current_user.space
+    product.space = current_space
+    db.session.commit()
+    
+    return({'msg':f'Updated products successfully'}), 200
```

Comment:
lol and this didnt throw a syntax error at you when you tested - abi you didn't even test it before sending your PR, should be:
```suggestion
    return make_response(jsonify({'msg':f'Updated products successfully'}), 200)
```